engine/runtime-instrument-common/src/main/java/org/enso/interpreter/instrument/IdExecutionService.java

```diff
@@ -237,9 +232,31 @@ public FunctionCallInfo(FunctionCallInstrumentationNode.FunctionCall call) {
         }
       }
 
-      notAppliedArguments = collectNotAppliedArguments(call);
+      return new FunctionPointer(moduleName, typeName, functionName);
     }
 
+    public static int[] collectNotAppliedArguments(Function function) {
+      FunctionSchema functionSchema = function.getSchema();
+      Object[] preAppliedArguments = function.getPreAppliedArguments();
+      boolean isStatic = preAppliedArguments[0] instanceof Type;
```

Comment:
Is this reliable? We could have an argument which is a type but the function is not a static?

Reply:
Yes, to resolve the function the self argument has to be present. And at this point, the function is resolved.